From 83409f0eb9cac288436b6b53b9d290c8c416834d Mon Sep 17 00:00:00 2001
From: Martin Kampas <martin.kampas@tieto.com>
Date: Thu, 17 Apr 2014 14:51:17 +0200
Subject: [PATCH] obex-client-tool: Set the length header

At least 'put' to other device with obexd v0.48 does not work without
this.
---
 tools/obex-client-tool.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/tools/obex-client-tool.c b/tools/obex-client-tool.c
index 54dbcbb..7207c77 100644
--- a/tools/obex-client-tool.c
+++ b/tools/obex-client-tool.c
@@ -135,11 +135,19 @@ static void cmd_put(int argc, char **argv)
 		return;
 	}
 
+	struct stat st;
+	if (fstat(fd, &st) < 0) {
+		close(fd);
+		g_printerr("fstat: %s\n", strerror(errno));
+		return;
+	}
+
 	data = g_new0(struct transfer_data, 1);
 	data->fd = fd;
 
 	g_obex_put_req(obex, put_data_cb, transfer_complete, data, &err,
 						G_OBEX_HDR_NAME, argv[1],
+						G_OBEX_HDR_LENGTH, st.st_size,
 						G_OBEX_HDR_INVALID);
 	if (err != NULL) {
 		g_printerr("put failed: %s\n", err->message);
-- 
1.9.2

